### PR TITLE
Use workspace dependencies more.

### PR DESCRIPTION
**Pull request by waywardmonkeys**
**Description**
In the past, workspace dependencies couldn't be used everywhere due to complications in the Firefox environment. These are believed to be resolved and so we can try using them again to reduce version management overhead and duplicated information.

This patch makes some dependencies into workspace dependencies.

The `bitflags` dependency is changed to version `2.6` to match what was in naga rather than the bare `2` used elsewhere.

**Testing**
Compilation.

**Checklist**

- [x] Run `cargo fmt`.
- [x] Run `cargo clippy`. If applicable, add:
  - [x] `--target wasm32-unknown-unknown`
  - [x] `--target wasm32-unknown-emscripten`
- [x] Run `cargo xtask test` to run tests.
- [X] ~~Add change to `CHANGELOG.md`. See simple instructions inside file.~~ EDIT by @ErichDonGubler: Not necessary. 🙂

### CODE REVIEW

**Comment by waywardmonkeys** [comment]
This could be a draft and get updated as more dependencies are switched over, or we could use this to be sure that there aren't issues from enabling this.

File: wgpu-hal/Cargo.toml
Comment (MarijnS95) [inline wgpu-hal/Cargo.toml]: These and everything below (outside of commentable diff range) are in the workspace table as well; any reason to _not_ inherit them?

Comment (waywardmonkeys) [reply wgpu-hal/Cargo.toml]: gpu-allocator is v0.26 in the workspace and I wanted to keep this as straight changes so far. (Nothing uses the v0.26 so there's no issue in updating it.)

I plan to do more, either as part of this PR or followups, but I wanted to make sure that there was no issue with me doing this to avoid potentially wasted effort.

**Comment by Wumpf** [comment]
looping in @ErichDonGubler to confirm we can have this since as pointed out [here](https://github.com/gfx-rs/wgpu/pull/6006#discussion_r1687345503) this used to be an issue for Mozilla's vendoring

File: wgpu-hal/Cargo.toml
Comment (MarijnS95) [reply wgpu-hal/Cargo.toml]: Just like you did for `bitflags`, let's bring the version in the workspace table in sync since this is the only user?

**Comment by teoxoy** [comment]
The was a bug with `cargo vendor` that required us to revert to non-inherited deps for the crates Firefox uses. It was fixed in rust 1.68 so we can now fully use workspace inheritance. https://github.com/gfx-rs/wgpu/issues/3407 is the bug tracking this.

**Comment by ErichDonGubler** [comment]
+1 to @teoxoy's comment; AFAIK, there are no blockers to merging this now.

File: wgpu-hal/Cargo.toml
Comment (ErichDonGubler) [reply wgpu-hal/Cargo.toml]: Yeah, let's make merge the 0.27 over to `Cargo.toml`, perhaps as a separate PR to not block this one? 👍🏻

**Comment by ErichDonGubler** [comment]
I'm going to merge this in a few hours, unless somebody has an objection (CC @gfx-rs/wgpu); I think the potential for conflicts is high, and I haven't felt or seen any controversy that _what's here already is fine by itself_.

**Review by ErichDonGubler — APPROVED** [review]

**Comment by waywardmonkeys** [comment]
I've rebased this forward, added in the further work that I'd done.

**Review by cwfitzgerald — APPROVED** [review]

**Comment by teoxoy** [comment]
Should this close https://github.com/gfx-rs/wgpu/issues/3407? I think so. I don't think we want all the deps in the workspace toml, only the ones present in more than one crate.

**Comment by MarijnS95** [comment]
> Should this close #3407? I think so. I don't think we want all the deps in the workspace toml, only the ones present in more than one crate.

To me it seems like tracking all versions in one coherent place is nice. That way you also don't have to be aware of this and go move dependencies back and forth when they start being used in >1 place, or have their use removed and end up remaining used by just one crate.

**Comment by teoxoy** [comment]
That's fair, let's keep it open for the time being then.

**Comment by ErichDonGubler** [comment]
Just waiting on @gfx-rs/naga's approval here.

**Review by teoxoy — APPROVED** [review]

**Pull request merged**
